### PR TITLE
Replace AMQP library

### PR DESCRIPTION
**Pull request by sonnysideup**
Using external library now that does not have race conditions causing a deadlock when closing the client during a a _reconnect event_.

### CODE REVIEW

**Comment by sonnysideup** [comment]
resolves #83

File: test/functional/helpers_test.go [hunk 1]
```diff
@@ -632,7 +633,9 @@ func testMessageDelivery(t *testing.T, ctx context.Context, client kubernetes.In
 		hostname = svc.Status.LoadBalancer.Ingress[0].IP
 	}
 	rmqURL := fmt.Sprintf("amqp://user:rabbitmq-password@%s:5672/", hostname)
-	conn, channel, err := amqp.Dial(rmqURL)
+	conn, err := amqp091.Dial(rmqURL)
+	require.NoError(t, err, "failed to connet to rabbitmq service")
```
Comment (steved) [inline test/functional/helpers_test.go:637]:
```suggestion
	require.NoError(t, err, "failed to connect to rabbitmq service")
```

File: pkg/messaging/amqp/publisher.go [hunk 2]
```diff
@@ -1,178 +0,0 @@
-package amqp
```
Comment (steved) [inline pkg/messaging/amqp/publisher.go:1]: Do you want to remove `manager.go` too?

**Review by steved — APPROVED** [review]

**Pull request merged**